ligra/undirected_edge.cc

```diff
@@ -0,0 +1,33 @@
+#include "ligra/undirected_edge.h"
+
+#include <algorithm>
+
+#include "pbbslib/utilities.h"
+
+namespace std {
+
+size_t hash<UndirectedEdge>::operator()(const UndirectedEdge& edge) const {
+  const uint64_t hash{pbbslib::hash64_2(edge.edge_.first)};
+  return hash ^ (pbbslib::hash64_2(edge.edge_.second)
+      + 0x9e3779b97f4a7c15 + (hash << 6) + (hash >> 2));
```

Comment:
What is this magic? Maybe we can link to http://xoshiro.di.unimi.it/splitmix64.c